pkg/scheduler/plugins/resource-strategy-fit/resource_strategy_fit.go

```diff
@@ -0,0 +1,220 @@
+	}()
+
+	nodeOrderFn := func(task *api.TaskInfo, node *api.NodeInfo) (float64, error) {
+		score := PlusScore(task, node, bp.weight)
+		klog.V(4).Infof("resourceStrategyFit score for Task %s/%s on node %s is: %v", task.Namespace, task.Name, node.Name, score)
+		return score, nil
+	}
+	if bp.weight.ResourceStrategyFitWeight != 0 {
+		ssn.AddNodeOrderFn(bp.Name(), nodeOrderFn)
+	} else {
+		klog.Infof("resourceStrategyFit weight is zero, skip node order function")
+	}
+}
+
+func PlusScore(task *api.TaskInfo, node *api.NodeInfo, weight ResourceStrategyFit) float64 {
+	score := 0.0
+	weightSum := 0
+	requested := task.Resreq
+	allocatable := node.Allocatable
+	used := node.Used
+
+	for _, resource := range requested.ResourceNames() {
+		request := requested.Get(resource)
+
+		allocate := allocatable.Get(resource)
+		nodeUsed := used.Get(resource)
+
+		_, found := weight.Resources[resource]
+		if !found {
+			continue
+		}
+
+		scoringType := weight.Resources[resource].Type
+		resourceWeight := weight.Resources[resource].Weight
+
+		var resourceScore float64
+		var err error
+		switch scoringType {
+		case config.MostAllocated:
+			resourceScore, err = mostRequestedScore(request, nodeUsed, allocate, resourceWeight)
+		case config.LeastAllocated:
+			resourceScore, err = leastRequestedScore(request, nodeUsed, allocate, resourceWeight)
+		}
+
+		if err != nil {
+			klog.V(4).Infof("task %s/%s cannot resourceStrategyFit node %s: resource: %s is %s, need %f, used %f, allocatable %f",
+				task.Namespace, task.Name, node.Name, resource, err.Error(), request, nodeUsed, allocate)
+			return 0
+		}
+
+		klog.V(5).Infof("task %s/%s on node %s resource %s, need %f, used %f, allocatable %f, weight %d, score %f",
+			task.Namespace, task.Name, node.Name, resource, request, nodeUsed, allocate, resourceWeight, resourceScore)
+
+		score += resourceScore
+		weightSum += resourceWeight
+	}
+
+	// mapping the result from [0, weightSum] to [0, 10(MaxPriority)]
+	if weightSum > 0 {
+		score /= float64(weightSum)
+	}
+	score *= float64(k8sFramework.MaxNodeScore * int64(weight.ResourceStrategyFitWeight))
+	return score
+}
+
+func mostRequestedScore(requested float64, used float64, capacity float64, weight int) (float64, error) {
+	if capacity == 0 || weight == 0 {
+		return 0, nil
+	}
+
+	usedFinally := requested + used
+	if usedFinally > capacity {
+		return 0, fmt.Errorf("node resources are not enough")
+	}
+
+	score := usedFinally * float64(weight) / capacity
+	return score, nil
+}
+
+func leastRequestedScore(requested float64, used float64, capacity float64, weight int) (float64, error) {
+	if capacity == 0 || weight == 0 {
+		return 0, nil
+	}
+
+	usedFinally := requested + used
+	if usedFinally > capacity {
+		return 0, fmt.Errorf("node resources are not enough")
+	}
+
+	score := (capacity - usedFinally) * float64(weight) / capacity
+	return score, nil
+}
+
+func (bp *resourceStrategyFitPlugin) OnSessionClose(ssn *framework.Session) {
```

Comment:
The naming rules of method callers need to be standardized. `bp` cannot see the relevance to resourceStrategyFitPlugin

Reply:
rsf done